R/MsBackendMzR.R

```diff
@@ -52,14 +52,110 @@ setMethod("readMsObject", signature(object = "MsBackendMzR",
                   rownames(data) <- NULL
                   object@spectraData <- DataFrame(data)
                   if (length(spectraPath) > 0) {
-                      old <- common_path(dataStorage(object))
-                      dataStoragePaths <- dataStorage(object)
-                      normalizedDataStoragePaths <- normalizePath(
-                          dataStoragePaths, winslash = "/", mustWork = FALSE)
-                      dataStorage(object) <- sub(old, spectraPath,
-                                                 normalizedDataStoragePaths)
+                      object <- .mz_backend_mzr_update_storage_path(
+                          object, spectraPath)
                   }
               }
               validObject(object)
               object
           })
+
+.mz_backend_mzr_update_storage_path <- function(x, spectraPath = character()) {
+    if (!length(x)) return(x)
+    old <- common_path(dataStorage(x))
+    dataStoragePaths <- dataStorage(x)
+    normalizedDataStoragePaths <- normalizePath(
+        dataStoragePaths, winslash = "/", mustWork = FALSE)
+    spectraPath <- normalizePath(spectraPath, winslash = "/", mustWork = FALSE)
+    x@spectraData$dataStorage <- sub(old, spectraPath,
+                                     normalizedDataStoragePaths)
+    x
+}
+
+################################################################################
+##
+## alabaster saveObject/readObject
+##
+################################################################################
+#' @importMethodsFrom alabaster.base saveObject
+#'
+#' @importFrom alabaster.base saveObjectFile
+#'
+#' @importFrom alabaster.base altSaveObject
+#'
+#' @exportMethod saveObject
+#'
+#' @rdname AlabasterParam
+setMethod("saveObject", "MsBackendMzR", function(x, path, ...) {
+    dir.create(path = path, recursive = TRUE, showWarnings = FALSE)
+    saveObjectFile(path, "ms_backend_mz_r",
+                   list(ms_backend_mz_r =list(version = "1.0")))
+    tryCatch({
+        do.call(altSaveObject,
+                list(x = x@spectraData, path = file.path(path, "spectra_data")))
+    }, error = function(e) {
+        stop("failed to save 'spectraData' of ", class(x)[1L], "\n - ",
+             e$message, call. = FALSE)
+    })
+    tryCatch({
+        do.call(altSaveObject,
+                list(x = x@peaksVariables,
+                     path = file.path(path, "peaks_variables")))
+    }, error = function(e) {
+        stop("failed to save 'peaksVariables' of ", class(x)[1L], "\n - ",
+             e$message)
+    })
+})
+
+#' @importFrom alabaster.base registerValidateObjectFunction
+#'
+#' @noRd
+validateAlabasterMzBackendMzR <- function(path = character(),
```

Comment:
That's a mouthfull aha, should it be `validateAlabasterM**s**BackendMzR` instead ?

Reply:
yes! thanks! - to many mz. and also good that this function is **not** exported - so users don't have to type that function name :D